nav2_pure_pursuit_controller/src/pure_pursuit_controller.cpp

```diff
@@ -238,59 +471,21 @@ PurePursuitController::transformGlobalPlan(
 }
 
 bool PurePursuitController::transformPose(
```

Comment:
This refactor of `transformPose` is good because it reduces the amount of parameters required.

But you removed a whole section of code related to "Transform data too old". Wat was the purpose of that code?

Reply:
No idea, I didn't write it. It might have been a result of a port before TF2 ROS2 had some of the API functions that are now available to deal with TF timeouts. DWB was one of the first things we ported to ROS2 so there's some legacy stuff floating around in that project more than others from primordial-ROS2 betas.

Reply:
Not sure, it was pulled from DWB which might have had that logic before TF2 in ROS2 was complete.